### PR TITLE
[RfR] APIv2: Filterable files

### PR DESCRIPTION
**Pull request by sloria**
### Changes

- Make NodeFilesList filterable
- Use httpretty rather than mocking requests.get for finer-grained control over mocked HTTP requests

### Ticket

https://openscience.atlassian.net/browse/OSF-4780

### CODE REVIEW

File: tests/api_tests/nodes/test_views.py [hunk 1]
```diff
@@ -2749,6 +2750,59 @@ def test_create_node_pointer_incorrect_type(self):
...
+    :param path: Waterbutler path, passed to waterbutler_api_url_for.
+    :param str method: HTTP method.
+    :param int status_code: HTTP status.
+    """
+    node = node
+    files = files or []
+    wb_url = waterbutler_api_url_for(node._id, provider=provider, path=path, meta=True)
+
+    default_file = {
+        u'contentType': None,
```
Comment (brianjgeiger) [inline tests/api_tests/nodes/test_views.py]: This is changing to a JSON API style response. I don't think it necessarily matters for what you're doing, but.

Comment (sloria) [inline tests/api_tests/nodes/test_views.py]: This was just taken from existing tests. We can change it when the WB responses are changed.

Comment (brianjgeiger) [inline tests/api_tests/nodes/test_views.py]: They are changed on staging now. I don't know if they're in production yet, but I presume so, cause release last night.

**Comment by brianjgeiger** [comment]
Seems to work well, no weird code. Approved with change to mock output to match new style. :penguin:

**Pull request merged**